server/drivers/amplitude/adapter.go

```diff
@@ -0,0 +1,439 @@
+package amplitude
+
+import (
+	"archive/zip"
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/jitsucom/jitsu/server/adapters"
+	"github.com/jitsucom/jitsu/server/drivers/base"
+	"github.com/jitsucom/jitsu/server/parsers"
+)
+
+const amplitudeURL = "https://amplitude.com"
+const amplitudeLayout = "20060102T15"
+
+const AmplitudeActiveUsers = "active_users"
+const AmplitudeAnnotations = "annotations"
+const AmplitudeAverageSessions = "average_sessions"
+const AmplitudeCohorts = "cohorts"
+const AmplitudeEvents = "events"
+const AmplitudeNewUsers = "new_users"
+
+const typeActiveUsers = "active"
+const typeNewUsers = "new"
+
+type AmplitudeAdapter struct {
+	httpClient *http.Client
+	authToken  string
+}
+
+func NewAmplitudeAdapter(apiKey, secretKey string, config *adapters.HTTPConfiguration) (*AmplitudeAdapter, error) {
+	httpClient := &http.Client{
+		Timeout: config.GlobalClientTimeout,
+		Transport: &http.Transport{
+			MaxIdleConns:        config.ClientMaxIdleConns,
+			MaxIdleConnsPerHost: config.ClientMaxIdleConnsPerHost,
+		},
+	}
+
+	token := fmt.Sprintf("%s:%s", apiKey, secretKey)
+	encodedToken := base64.StdEncoding.EncodeToString([]byte(token))
+	secretToken := fmt.Sprintf("Basic %s", encodedToken)
+
+	return &AmplitudeAdapter{
+		httpClient: httpClient,
+		authToken:  secretToken,
+	}, nil
+}
+
+func (a *AmplitudeAdapter) Close() error {
+	a.httpClient.CloseIdleConnections()
+	return nil
+}
+
+func (a *AmplitudeAdapter) GetStatus() error {
+	request := &adapters.Request{
+		URL:    amplitudeURL + "/status",
+		Method: "GET",
+	}
+
+	status, _, err := a.doRequest(request)
+	if err != nil {
+		return err
+	}
+
+	if status != http.StatusOK {
+		return fmt.Errorf("Status returns not OK")
+	}
+
+	return nil
+}
+
+func (a *AmplitudeAdapter) GetEvents(interval *base.TimeInterval) ([]map[string]interface{}, error) {
+	start := interval.LowerEndpoint().Format(amplitudeLayout)
+	end := interval.UpperEndpoint().Format(amplitudeLayout)
+
+	url := fmt.Sprintf("%v/api/2/export?start=%s&end=%s", amplitudeURL, start, end)
+
+	request := &adapters.Request{
+		URL:     url,
+		Method:  "GET",
+		Headers: map[string]string{},
+	}
+
+	request.Headers["Authorization"] = a.authToken
```

Comment:
It is politically incorrect because `doRequest` should only `do` but not `prepare` requests.
Especially, `getStatus` do not use the authorization header.
So, the authorization header is moved into request preparing.